rust/src/lib.rs

```diff
@@ -107,12 +113,44 @@ unsafe fn wallet_db(
 /// - The memory referenced by `fsblock_db` must not be mutated for the duration of the function call.
 /// - The total size `fsblock_db_len` must be no larger than `isize::MAX`. See the safety
 ///   documentation of pointer::offset.
-fn block_db(fsblock_db: *const u8, fsblock_db_len: usize) -> Result<FsBlockDb, failure::Error> {
+fn block_db(fsblock_db: *const u8, fsblock_db_len: usize) -> anyhow::Result<FsBlockDb> {
     let cache_db = Path::new(OsStr::from_bytes(unsafe {
         slice::from_raw_parts(fsblock_db, fsblock_db_len)
     }));
     FsBlockDb::for_path(cache_db)
-        .map_err(|e| format_err!("Error opening block source database connection: {}", e))
+        .map_err(|e| anyhow!("Error opening block source database connection: {}", e))
+}
+
+/// Initializes global Rust state, such as the logging infrastructure and threadpools.
+///
+/// # Panics
+///
+/// This method panics if called more than once.
+#[no_mangle]
+pub extern "C" fn zcashlc_init_on_load() {
+    // Set up the tracing layer for the Apple OS logging framework.
+    let os_layer = os_log::layer("co.electriccoin.ios", "rust").with_filter(LevelFilter::INFO);
```

Comment:
Unfortunately I'm not seeing this reverse domain in any of the fields of the events reported by OS Logger.